include/util/exception.hpp

```diff
@@ -40,8 +41,14 @@ namespace util
 class exception final : public std::exception
 {
   public:
-    explicit exception(const char *message) : message(message) {}
-    explicit exception(std::string message) : message(std::move(message)) {}
+    explicit exception(const char *message_) : message(message_)
+    {
+        util::SimpleLogger().Write(logERROR) << message;
```

Comment:
Yeah throwing while there's an exception active is a bad idea. That's the reason your dtor must not throw, too.